wzdx/combination/attenuator.py

```diff
@@ -0,0 +1,190 @@
+    start = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def get_route_info_geotab(geotab):
+    start = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, (datetime)):
+        return obj.strftime(ISO_8601_FORMAT_STRING)
+    raise TypeError("Type %s not serializable" % type(obj))
+
+
+def main():
+    with open('./wzdx/sample_files/raw/geotab_avl/geotab_all.json') as f:
+        geotab_avl = json.loads(f.read())
+    with open('./wzdx/sample_files/enhanced/planned_events/planned_event_all.json') as f:
+        planned_event = json.loads(f.read())
+
+    combined_events = get_combined_events(geotab_avl, planned_event)
+
+    with open('./wzdx/sample_files/enhanced/planned_events/planned_event_combined.json', 'w+') as f:
+        f.write(json.dumps(combined_events, indent=2))
+
+
+def validate_directionality(geotab, planned_event):
+    geotab_bearing = geotab['avl_location']['position']['bearing']
+    planned_event_direction = planned_event['features'][0]['properties']['core_details']['direction']
+
+    geotab_direction = polygon_tools.get_closest_direction_from_bearing(
+        geotab_bearing, planned_event_direction)
+
+    print(geotab_direction, planned_event_direction)
+
+    return geotab_direction == planned_event_direction
+
+
+def get_combined_events(gebtab_msgs, planned_events):
+    conmbined_events = []
+    for i in identify_overlapping_features(gebtab_msgs, planned_events):
+        if not validate_directionality(*i):
+            logging.info(
+                "Ignoring matching Geotab message because the direction does not match the planned event")
+            continue
+        feature = combine_geotab_with_planned_event(*i)
+        wzdx = i[1]
+        wzdx['features'] = [feature]
+        conmbined_events.append(wzdx)
+    return conmbined_events
+
+
+def identify_overlapping_features(gebtab_msgs, planned_events):
+    geotab_routes = {}
+    matching_routes = []
+
+    for gebtab_msg in gebtab_msgs:
+        # assume 1 feature per wzdx planned_event
```

Comment:
This assumption doesn't really apply here, was it just left over from code below?

Reply:
removed